### PR TITLE
password hidden while entering

### LINKED ISSUE #216: Password is not hiding while entering the password
#### Software and hardware versions 
Android Samsung j8

#### Problem description
While we entering the password the text should i.e it will show dots after entering each character

#### Steps to reproduce the problem
go to settings and enable set password and tap on set hotspot password and enter anything observe the behavior

#### Expected behavior
while entering the password it should hide after each character

### PR DESCRIPTION
**Pull request by rob729**
Closes #216 

#### What has been done to verify that this works as intended?
Tested the app on my device(Lenovo P2) and the issue is solved now.

#### Why is this the best possible solution? Were any other approaches considered?
This is best possible solution as it gets the work done with minimum addition in the lines of code.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
The passwords remains hidden after user has entered it.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkCode` and confirmed all checks still pass OR confirm CircleCI build passes
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/share/blob/master/share_app/src/main/assets/open_source_licenses.html).

### CODE REVIEW

**Comment by rob729** [comment]
@shobhitagarwal1612 @lakshyagupta21 Please review this PR.

**Comment by shobhitagarwal1612** [comment]
Now that the password is hidden, there should be a button to check the entered password for users to be able to check when needed

**Comment by rob729** [comment]
@shobhitagarwal1612 There is no option to allow the toggling of visibility of text in EditTextPreference.
So what should I do now?

**Comment by shobhitagarwal1612** [comment]
Have a look at TextInputEditText widget

**Comment by rob729** [comment]
I don't think it will work with Shared Preferences.

**Comment by shobhitagarwal1612** [comment]
You can create a custom dialog with the above widget and then save the password on OK button press.

**Comment by Chromicle** [comment]
> @shobhitagarwal1612 There is no option to allow the toggling of visibility of text in EditTextPreference.
> So what should I do now?

Yeah  Toggling is not working but you can make checkbox like show password below the input of password

**Comment by shobhitagarwal1612** [comment]
@Chromicle Falling back to checkbox would be ok if there was no other option. But the above alternate I mentioned is the standard way to implement it. So, it makes sense to try that out first

**Comment by Chromicle** [comment]
> @Chromicle Falling back to checkbox would be ok if there was no other option. But the above alternate I mentioned is the standard way to implement it. So, it makes sense to try that out first

Yeah making cheak box instead of toogle button is not at all good but when I tried to put the toogle button it does not give any warning but the toggle but not appearing so that is what the reason I suggested for cheak box 
@rob729 while you are doing this PR for odk directory make onother preference content like storage instead of Hostspot settings

**Comment by lakshyagupta21** [comment]
> I tried to put the toogle button it does not give any warning but the toggle but not appearing so that is what the reason I suggested for cheak box

I didn't understand what you're trying to say. Can you rephrase it and explain better.


> @rob729 while you are doing this PR for odk directory make onother preference content like storage instead of Hostspot settings

And the scope of this issue is just to hide character on each new character entry in the field. Lets create separate issue for this and resolve in new PR. 
@rob729 Thanks for taking up this issue, I would suggest you to try `TextInputEditText` as suggested by @shobhitagarwal1612 , if you face any difficulty in implementing it let us know.

**Comment by lakshyagupta21** [comment]
@rob729 You still working on this?

**Comment by rob729** [comment]
@lakshyagupta21 No

**Comment by lakshyagupta21** [comment]
Closing this for now, let someone else take this up then.
